core/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -120,8 +121,11 @@ protected boolean validate(ClusterState newState) {
         AsyncSingleAction(Task task, Request request, ActionListener<Response> listener) {
             this.task = task;
             this.request = request;
-            if (task != null) {
-                request.setParentTask(clusterService.localNode().getId(), task.getId());
+            DiscoveryNode discoveryNode = clusterService.localNode();
+            if (task != null && discoveryNode != null) {
```

Comment:
@jasontedor Thanks for noticing, forgot the else if (discoveryNode == null)

I will update the pull request with your recommended way of checking this in the ClusterService.